Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVirtualPortGroup.ps1

```diff
@@ -0,0 +1,236 @@
+
+Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+[DscResource()]
+class VMHostVirtualPortGroup : VMHostNetworkBaseDSC {
+    <#
+    .DESCRIPTION
+
+    Specifies the name for the Port Group.
+    #>
+    [DscProperty(Key)]
+    [string] $PortGroupName
+
+    <#
+    .DESCRIPTION
+
+    Specifies the Virtual Switch associated with the Port Group.
+    The Virtual Switch must be a Standard Virtual Switch.
+    #>
+    [DscProperty(Mandatory)]
+    [string] $VirtualSwitch
+
+    <#
+    .DESCRIPTION
+
+    Value indicating if the Port Group should be Present or Absent.
+    #>
+    [DscProperty(Mandatory)]
+    [Ensure] $Ensure
+
+    <#
+    .DESCRIPTION
+
+    Specifies the VLAN ID for ports using this Port Group. The following values are valid:
+    0 - specifies that you do not want to associate the Port Group with a VLAN.
+    1 to 4094 - specifies a VLAN ID for the Port Group.
+    4095 - specifies that the Port Group should use trunk mode, which allows the guest operating system to manage its own VLAN tags.
+    #>
+    [DscProperty()]
+    [nullable[int]] $VLanId
+
+    hidden [int] $VLanIdMaxValue = 4095
+
+    [void] Set() {
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        $foundVirtualSwitch = $this.GetVirtualSwitch($vmHost, $this.VirtualSwitch)
+        $portGroup = $this.GetVirtualPortGroup($vmHost, $foundVirtualSwitch)
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            if ($null -eq $portGroup) {
+                $this.AddVirtualPortGroup($foundVirtualSwitch)
+            }
+            else {
+                $this.UpdateVirtualPortGroup($portGroup)
+            }
+        }
+        else {
+            if ($null -ne $portGroup) {
+                $this.RemoveVirtualPortGroup($portGroup)
+            }
+        }
+    }
+
+    [bool] Test() {
+        $this.ConnectVIServer()
```

Comment:
Do we call DisconnectVIServer somewhere? If we make connect on each resource method without disconnect would be harmful for the server

Reply:
Disconnect logic will be introduced in a separate PR.